sol-anchor/idl.go

```diff
@@ -0,0 +1,612 @@
+package solanchor
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"unicode"
+
+	"github.com/golang-cz/textcase"
+)
+
+type IDL struct {
+	Events       []Event       `json:"events"`
+	Instructions []Instruction `json:"instructions"`
+	Metadata     Metadata      `json:"metadata"`
+	Types        []Type        `json:"types"`
+}
+
+type Metadata struct {
+	Address string `json:"address"`
+}
+
+// --- EVENTS
```

Comment:
```suggestion
// --- EVENTS
```
Instead of having everything in a file, I would rather used multiple files to improve code readability. For example, you create an `events.go`, `instructions.go` within an `idl` folder...

And write all the `toSnakeCase` kind of functions in a `utils.go` to share the code among all those files...